src/components/biddingPage/dr/listItem.tsx

```diff
@@ -1,178 +1,158 @@
-
-interface IData {
-  date: string;
-  interval: string;
-  time: number;
-  value: number;
-  price: number;
-  total: number;
-  status: string;
-  accepted: boolean;
-}
 
 interface IProps {
+  start_time: string;
+  end_time: string | null;
+  uuid: string;
+  accepted: Boolean | null;
   date: string;
   interval: string;
-  time: number;
   value: number;
   price: number;
   total: number;
   status: string;
-  accepted: boolean;
   isAggr: boolean;
-  data: IData[];
-  setData(d: IData[]): void;
 }
 
 const ListItem: React.FC<IProps> = ({
+  start_time,
+  end_time,
+  uuid,
+  accepted,
   date,
   interval,
-  time,
   value,
   price,
   total,
-  status,
   isAggr,
 }) => {
   // i18n
   const { t } = useTranslation();
 
-  // display data
-  const [displayDate, setDisplayDate] = useState<string>(date);
-  const [displayInterval, setDisplayInterval] = useState<string>(interval);
-  const [displayValue, setDisplayValue] = useState<number>(value);
-  const [displayPrice, setDisplayPrice] = useState<number>(price);
-  const [displayTotal, setDisplayTotal] = useState<number>(total);
-
-  // edit mode
-  const [editMode, setEditMode] = useState<boolean>(false);
-
-  // deleted
-  const [deleted, setDeleted] = useState<boolean>(false);
-
-  // bid btn disabled or not
-  const [DeleteBtnDisabled, setDeleteBtnDisabled] = useState<boolean>(false);
+  // accept bid btn disabled or not
+  const [btnDisabled, setBtnDisabled] = useState<boolean>(!!accepted);
 
   // bid btn's text
-  const [DeleteBtnText, setDeleteBtnText] = useState<string>(
-    t('biddingpage.bid'),
+  const [btnText, setBtnText] = useState<string>(
+    accepted ? t('biddingpage.accepted') : t('biddingpage.accept'),
   );
 
-  // this bid is editable or not
-  const [editable, setEditable] = useState<boolean>(true);
+  // aggregator clicked accept button
+  const [acceptClicked, setAcceptClicked] = useState<boolean>(false);
 
-  // current time
-  const [currDate, setCurrDate] = useState<Date>(new Date());
+  // start hour
+  const [startHr, setStartHr] = useState<number>(0);
 
-  useEffect(() => {
-    if (isAggr) {
-      setDeleteBtnText(t('biddingpage.accept'));
-      setDeleteBtnDisabled(false);
-    } else if (
-      new Date().getTime() >=
-        new Date(
-          new Date().getFullYear(),
-          new Date().getMonth(),
-          new Date().getDate(),
-          10,
-          30,
-        ).getTime() &&
-      date === dayjs().add(1, 'day').format('YYYY/MM/DD')
-    ) {
-      // expired
-      setEditable(false);
-      setDeleteBtnDisabled(true);
-    } else {
-      // available
-      setEditable(true);
-      setDeleteBtnDisabled(false);
-    }
-  }, [isAggr, date, time]);
+  // end hour
+  const [endHr, setEndHr] = useState<number>(0);
 
-  // determine editable or not every minute
-  useEffect(() => {
-    setInterval(() => setCurrDate(new Date()), 1000);
-  }, []);
+  // hour array
+  const hrArr = [
```

Comment:
You can import this array from `constant.ts`

Reply:
Done. Thanks.